Utils/src/DeepAK8Producer.cc

```diff
@@ -0,0 +1,73 @@
+// system include files
+#include <memory>
+#include <vector>
+#include <string>
+#include <iterator>
+#include <algorithm>
+#include <TMath.h>
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+// new includes
+#include "NNKit/FatJetNN/interface/FatJetNN.h"
+#include "NNKit/FatJetNN/interface/FatJetNNHelper.h"
+
+#include "TLorentzVector.h"
+
+class DeepAK8Producer : public edm::global::EDProducer<> {
+public:
+    explicit DeepAK8Producer(const edm::ParameterSet&);
+private:
+    void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+    edm::EDGetTokenT<edm::View<pat::Jet>> JetAK8Tok_;
+
+    deepntuples::FatJetNN* fatjetNN_;
+};
+
+DeepAK8Producer::DeepAK8Producer(const edm::ParameterSet& iConfig) :
+    JetAK8Tok_(consumes<edm::View<pat::Jet>>(iConfig.getUntrackedParameter<edm::InputTag>("JetDeepAK8", edm::InputTag("slimmedJetsAK8")))),
+    fatjetNN_(nullptr)
+{
+    // Initialize the FatJetNN class in the constructor
+    auto cc = consumesCollector();
+    fatjetNN_ = new deepntuples::FatJetNN(iConfig, cc);
+    // Load json for input variable transformation
+    fatjetNN_->load_json("preprocessing.json"); // use the full path or put the file in the current working directory (i.e., where you run cmsRun)
+    // Load DNN model and parameter files
+    fatjetNN_->load_model("resnet-symbol.json", "resnet.params"); // use the full path or put the file in the current working directory (i.e., where you run cmsRun)
+}
+
+void DeepAK8Producer::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const
+{
+    // Need to access event info
+    fatjetNN_->readEvent(iEvent, iSetup);
+
+    // Get, loop over the jets, and fill  
+    edm::Handle<edm::View<pat::Jet>> jetDeepAK8;
+    iEvent.getByToken(JetAK8Tok_, jetDeepAK8);
+
+    auto tDiscriminatorDeep = std::make_unique<std::vector<double>>();
```

Comment:
It would be better to produce these as userfloats. For an example, see https://github.com/TreeMaker/TreeMaker/blob/Run2_2017/Utils/src/JetUncertaintyProducer.cc (follow the `produces<edm::ValueMap<float>>("");` part).

Then, you can add this module to the python config after the double b-tagger, and update the jet collection to know about the new userfloats by modifying the `addJetInfo` call:
https://github.com/TreeMaker/TreeMaker/blob/0a9d941d505822f5537a0b6fa67a3e5fa07d15ad/TreeMaker/python/makeTreeFromMiniAOD_cff.py#L731